pytensor/link/pytorch/dispatch/scalar.py

```diff
@@ -0,0 +1,328 @@
+
+    func_name = nfunc_spec[0]
+    print
+    # if "." in func_name:
+    #     pytorch_func = functools.reduce(getattr, [jax, *func_name.split(".")])
+    # else:
+    pytorch_func = getattr(torch, func_name)
+
+    if len(node.inputs) > op.nfunc_spec[1]:
+        # Some Scalar Ops accept multiple number of inputs, behaving as a variadic function,
+        # even though the base Op from `func_name` is specified as a binary Op.
+        # This happens with `Add`, which can work as a `Sum` for multiple scalars.
+        pytorch_variadic_func = getattr(torch, op.nfunc_variadic, None)
+        if not pytorch_variadic_func:
+            raise NotImplementedError(
+                f"Dispatch not implemented for Scalar Op {op} with {len(node.inputs)} inputs"
+            )
+
+        def pytorch_func(*args):
+            return pytorch_variadic_func(
+                torch.stack(torch.broadcast_tensors(*args), axis=0), axis=0
+            )
+
+    return pytorch_func
+
+
+@functools.singledispatch
+def pytorch_funcify_scalar_op_via_py_operators(op):
+    """Specialized JAX dispatch for Elemwise operations where all inputs are Scalar arrays.
+
+    Scalar (constant) arrays in the JAX backend get lowered to the native types (int, floats),
+    which can perform better with Python operators, and more importantly, avoid upcasting to array types
+    not supported by some JAX functions.
+    """
+    return None
+
+
+@pytorch_funcify_scalar_op_via_py_operators.register(Add)
+def pytorch_funcify_scalar_Add(op):
+    def elemwise(*inputs):
+        return sum(inputs)
+
+    return elemwise
+
+
+@pytorch_funcify_scalar_op_via_py_operators.register(Mul)
+def pytorch_funcify_scalar_Mul(op):
+    import operator
+    from functools import reduce
+
+    def elemwise(*inputs):
+        return reduce(operator.mul, inputs, 1)
+
+    return elemwise
+
+
+@pytorch_funcify_scalar_op_via_py_operators.register(Sub)
+def pytorch_funcify_scalar_Sub(op):
+    def elemwise(x, y):
+        return x - y
+
+    return elemwise
+
+
+@pytorch_funcify_scalar_op_via_py_operators.register(IntDiv)
+def pytorch_funcify_scalar_IntDiv(op):
+    def elemwise(x, y):
+        return x // y
+
+    return elemwise
+
+
+@pytorch_funcify_scalar_op_via_py_operators.register(Mod)
+def pytorch_funcify_scalar_Mod(op):
+    def elemwise(x, y):
+        return x % y
+
+    return elemwise
+
+
+@pytorch_funcify.register(Cast)
+def pytorch_funcify_Cast(op, **kwargs):
+    def cast(x):
+        return torch.tensor(x).astype(op.o_type.dtype)
+
+    return cast
+
+
+@pytorch_funcify.register(Identity)
+def pytorch_funcify_Identity(op, **kwargs):
+    def identity(x):
+        return x
+
+    return identity
+
+
+@pytorch_funcify.register(Clip)
+def pytorch_funcify_Clip(op, **kwargs):
+    """Register the translation for the `Clip` `Op`.
+
+    """
+
+    def clip(x, min, max):
+        return torch.where(x < min, min, torch.where(x > max, max, x))
```

Comment:
`torch.clip`?

Reply:
Similar to JAX, torch also gives a different result when `min>max`

https://github.com/pymc-devs/pytensor/blob/d80c0bf7bfa5d6b9812ef6381cfecb6b9bf948a6/pytensor/link/jax/dispatch/scalar.py#L191-L196

An example:
```
x = pt.tensor('x', shape=(2,2))
fn = function([x], [pt.clip(x, 20, 10)])
test = 40*torch.rand(2,2)
print(f'Original:\n{test.numpy()}\n\n')
print(f'PyTensor (pt.clip):\n{fn(test.numpy())}\n\n')
print(f'Pytorch (torch.clip):\n{torch.clip(test, 20, 10)}')
```
The output for this would be:
```
Original:
[[ 0.23448467  7.9555655 ]
 [37.57132    13.594839  ]]


PyTensor (pt.clip):
[array([[20., 20.],
       [10., 20.]])]


Pytorch (torch.clip):
tensor([[10., 10.],
        [10., 10.]])
```

Reply:
I have no idea what PyTensor is doing, we should do the same as Numpy which is also what torch is outputting